PIL/TiffImagePlugin.py

```diff
@@ -1737,17 +1737,28 @@ def fixOffsets(self, count, isShort=False, isLong=False):
                 self.rewriteLastLong(offset)
 
 def _save_all(im, fp, filename):
-    if not hasattr(im, "n_frames"):
+    encoderinfo = im.encoderinfo.copy()
+    encoderconfig = im.encoderconfig
+    append_images = encoderinfo.get("append_images", [])
+    if not hasattr(im, "n_frames") and not len(append_images):
         return _save(im, fp, filename)
 
     cur_idx = im.tell()
     try:
         with AppendingTiffWriter(fp) as tf:
-            for idx in range(im.n_frames):
-                im.seek(idx)
-                im.load()
-                _save(im, tf, filename)
-                tf.newFrame()
+            for ims in [im]+append_images:
+                ims.encoderinfo = encoderinfo
+                ims.encoderconfig = encoderconfig
+                if not hasattr(ims, "n_frames"):
+                    nfr = 1
+                else:
+                    nfr = ims.n_frames
```

Comment:
Can we do this in terms of just appending all frames of the current image to the new image? Getting the number of frames (at least in some cases) involves processing the entire image twice (namely, GIFs) while throwing away the results.